### PR TITLE
pinctrl: some minor updates

### PR DESCRIPTION
**Pull request by gmarull**
- pinctrl: require ; after PINCTRL_DT_(INST_)DEFINE macros
  The PINCTRL_DT_(INST_)DEFINE macros already defined the trailing ;,
  making its usage inconsistent with other macros such as DEVICE_DT_DEFINE.
- pinctrl: use groupN instead of pinsN for groups
  The pinsN group name can be a confusing in some circumstances, so change
  it to groupN. Some platforms (e.g. nrf or gd32) are already using
  groupN. Documentation and API tests have been updated.

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]
Not sure why the last commit is not folded into the previous one but otherwise LGTM.

**Comment by gmarull** [comment]
> Not sure why the last commit is not folded into the previous one but otherwise LGTM.

squashed

**Review by nandojve — APPROVED** [review]

File: drivers/serial/uart_nrfx_uarte.c [hunk 1]
```diff
@@ -1983,7 +1983,7 @@ static int uarte_nrfx_pm_action(const struct device *dev,
 #define UART_NRF_UARTE_DEVICE(idx)					       \
 	UARTE_INT_DRIVEN(idx);						       \
 	UARTE_ASYNC(idx);						       \
-	IF_ENABLED(CONFIG_PINCTRL, (PINCTRL_DT_DEFINE(UARTE(idx))))	       \
+	IF_ENABLED(CONFIG_PINCTRL, (PINCTRL_DT_DEFINE(UARTE(idx))));	       \
```
Comment (anangl) [inline drivers/serial/uart_nrfx_uarte.c:1986]: For consistency, this should be actually:
```suggestion
	IF_ENABLED(CONFIG_PINCTRL, (PINCTRL_DT_DEFINE(UARTE(idx));))	       \
```
but even with this extra semicolon for !PINCTRL, it is acceptable.

**Review by anangl — APPROVED** [review]

**Review by erwango — APPROVED** [review]

File: drivers/serial/uart_nrfx_uarte.c (hunk 1 above)
Comment (gmarull) [reply drivers/serial/uart_nrfx_uarte.c]: thanks, adjusted

**Pull request merged**